internal/gitutil/gitutil.go

```diff
@@ -378,20 +394,27 @@ loop:
 		// commit sha.
 		validFullSha := s == strings.TrimSpace(rr.Stdout)
 		_, resolved := gur.ResolveRef(s)
-
+		// fetchedRefWithURI is the cache key created from repo URI SHA and ref
+		fetchedRefWithURI := fmt.Sprintf("%s-%s", uriSha, s)
```

Comment:
we should probably reverse these `ref-SHA` (`main-sldksls`, `main-hkdldl`)

Reply:
URI SHA is confusing with the `git SHA`. May be let's use `repoCacheKey` term for `URI SHA` and fetchedRefWithURI can be renamed to `refCacheKey`.

Reply:
@droot based on our conversation i did some digging. https://github.com/GoogleContainerTools/kpt/blob/72ef5b29cb372ed1d28e1380782651f366e2d14f/internal/gitutil/gitutil.go#L301-L307
The current `uriSHA` was an md5 of GitUpstreamRepo.URI so it's the same for each ref. I have refactored to just use the ref as the cache key instead.